src/lib/store.ts

```diff
@@ -43,12 +59,21 @@ export class NaiveStore implements Store {
     this.users = new InMemorySparseMap<string, User>(
       (user: string, api: Api) => api.users.info({user}).map((x: any) => x.user! as User).toPromise(),
       'merge');
-    this.messages = new InMemorySparseMap<string, Array<Message>>((channel, api: Api) => {
-      return api.channels.history({ channel }).map(({ messages }: { messages: Array<Message> }) => {
-        messages.api = api;
-        return messages;
+
+    this.messages = new InMemorySparseMap<MessagesKey, MessageCollection>((key: MessagesKey, api: Api) => {
```

Comment:
I'm not smart enough to frame this problem elegantly in the model, so I've just done the most naïve thing possible.